src/functional/map.go

```diff
@@ -0,0 +1,11 @@
+package functional
+
+func Map[In any, Out any](arr []In, transformFn func(it In) Out) []Out {
```

Comment:
my changes for this and `Filter` were more about the parameter / variable names and less about the whitespace